upgrade/1.2/Stage_0_Prerequisites.md

```diff
@@ -107,20 +108,128 @@ backup of Workload Manager configuration data and files is created. Once complet
+
+   During the CSM 1.2 upgrade, UANs will retain their pre-1.2 CAN IPs in order to minimize disruption to UANs. However,
+   toward the end of the CSM 1.2 upgrade, UANs will stop registering themselves on CMN and will receive new IPs on the
+   CAN/CHN network below. This process is described in more detail in [UAN Migration](#uan-migration).
+
+1. Customer Access Network (CAN) / Customer High-speed Network (CHN)
+
+   For user traffic only (e.g. users running and monitoring jobs), CSM 1.2 allows you to pick one of two networks:
+
+      - Customer Access Network (CAN): this is a new network (VLAN6 in switches) that runs over the management network. This
+         network must not be confused with pre-1.2 CAN, which was a monolithic network that allowed both user and administrative
+         traffic, was configured as VLAN7 in switches, and is now renamed to CMN. The new CAN allows only user traffic.
+
+      - Customer High-speed Network (CHN): this is a new network (VLAN5 in switches) that runs over the high-speed fabric.
+
+   You must only pick either the new CAN or CHN, but not both. The rest of the installation guide will provide you with
+   options for configuring either the new CAN or CHN.
+
+Pivoting the pre-1.2 CAN to the new CMN allows administrative traffic (already on the pre-1.2 CAN) to remain as-is while
+moving standard user traffic to a new site-routable network.
+
+### UAN migration
+
+Certain steps are taken in order to minimize disruption to UANs during the CSM 1.2 upgrade process:
+
+1. During the upgrade, the switch `1.2 Preconfig` will not remove UAN ports from the CMN VLAN (the pre-1.2 CAN), allowing UANs
+   to retain their existing IPs during the CSM 1.2 upgrade process. Traffic to and from UANs will still flow through CMN, but
+   may also flow through CAN/CHN networks if desired.
+
+1. CFS will be temporarily disabled for UANs, so that running CFS plays does not remove CMN interfaces from UANs. As mentioned
+   in [Abstract (Stage 0.3)](#abstract-stage-03), network configuration is controlled by data in SLS, but CFS plays also pick up
+   the same SLS data, which can lead to UANs being prematurely removed from CMN and causing UAN outage. As such, CFS plays
+   need to be disabled for UANs.
+
+   To disable CFS plays for UANs, you must remove CFS assignment for UANs by running the following command:
+
+   ```bash
+   what-is-the-command-to-disable-CFS-for-UANs.sh
+   ```
+
+1. UAN reboots must be avoided. Rebooting a UAN can re-enable CFS and can ultimately lead to removing CMN interfaces from UANs.
+   As a system administrator, you must inform your users to avoid UAN reboots during the CSM 1.2 upgrade process. If, however,
+   a UAN is rebooted, you can run the following set of commands to add the CMN interface back on the UAN:
+
+   ```bash
+   what-is-the-command-to-readd-CMN-to-UANs-after-accidental-reboot.sh
```

Comment:
Here is the diff in the UAN cfs play we worked out for this recovery scenario
```
--- a/roles/uan_interfaces/tasks/can-v2.yml
+++ b/roles/uan_interfaces/tasks/can-v2.yml
@@ -33,21 +33,16 @@
 - name: Get Customer Access Network info from SLS
   local_action:
     module: uri
-    url: "http://cray-sls/v1/search/networks?name={{ sls_can_name }}"
+    url: "http://cray-sls/v1/search/networks?name=CMN"
     method: GET
   register: sls_can

-- name: Get Customer Access Network CIDR from SLS, if network exists.
-  # This assumes that the CAN network is _always_ the third item in the array. This makes the
-  # implementation fragile. See CASMCMS-6714.
-  set_fact:
-    customer_access_network: "{{ sls_can.json[0].ExtraProperties.Subnets[2].CIDR }}"
-  when: sls_can.status == 200
-
-- name: Get Customer Access Network Gateway from SLS, if network exists
-  set_fact:
-    customer_access_gateway: "{{ sls_can.json[0].ExtraProperties.Subnets[2].Gateway }}"
-  when: sls_can.status == 200

+- name: "Get {{ uan_user_access_cfg | upper }} CIDR from SLS, if network exists."
+  set_fact:
+    customer_access_network: "{{ item.CIDR }}"
+    customer_access_gateway: "{{ item.Gateway }}"
+  loop: "{{ sls_can.json[0].ExtraProperties.Subnets }}"
+  when: item.FullName == "CMN Bootstrap DHCP Subnet"
```

Reply:
Thanks @alanm-hpe. I have added that patch file as well as a comment to go read the Configuration Management docs in how to apply that patch file to the `vcs/cray/uan-config-management.git` repository. Please review and feel free to reopen this comment if you think something more needs to be done.